File: fenicia-web/src/types/basic/order/create-order-response.ts
import { OrderStatus } from "./order-status";
import { PaymentMethod } from "./payment-method";


export type CreateOrderResponse = {
    id: string;
    orderNumber: string;
    userId: string;
    customerId: string;
    totalAmount: number;
    discountAmount: number;
    totalQuantity: number;
    saleDate: string;
    status: OrderStatus;
    paymentMethod: PaymentMethod;
    notes?: string;
    employeeId?: string;
};

File: Src/APIs/Fenicia.Module.Basic/Domains/Order/DTOs/CreateOrderResponse.cs
using System.ComponentModel.DataAnnotations;
using Fenicia.Common.Enums.Auth;
using Fenicia.Common.Enums.Basic;

namespace Fenicia.Module.Basic.Domains.Order.DTOs;

public record CreateOrderResponse(
    [Required] Guid Id,
    [Required][MaxLength(200)] string OrderNumber,
    [Required] Guid UserId,
    [Required] Guid CustomerId,
    decimal TotalAmount,
    decimal DiscountAmount,
    int TotalQuantity,
    [Required] DateTime SaleDate,
    [Required] OrderStatus Status,
    [Required] PaymentMethod PaymentMethod,
    string? Notes = null,
    Guid? EmployeeId = null);
